### DISCUSSION TITLE
httpx equivalent to pip-system-certs?

### DISCUSSION THREAD

**Discussion by gbeane** [encode/httpx#3027] [Q&A] [answered]
Our corporate IT requires that we run an application called ZScaler on our laptops.  This causes SSL certificate validation to fail unless we use the system certificate store or add the ZScaler root certificate to the Python certify package's cacert.pem file.

Installing pip-system-certs wraps the requests package so it will use the system certificates, which fixes the issue with ZScaler when using requests. However, since pip-system-certs only wraps requests it doesn't fix SSL issues when using httpx as a replacement to requests.

https://gitlab.com/alelec/pip-system-certs/-/blob/master/pip_system_certs/wrapt_requests.py?ref_type=heads

I am wondering if there is something similar that wraps httpx so that it will use system certificates.  For now I need to manually add the ZScaler certificate to Lib/site-packages/certifi/cacert.pem in my virtualenv, but I don't want other corporate users of my package to need to do this every time they create a new virtualenv

**Comment by zanieb** [comment, marked as answer]
Perhaps you're looking for https://github.com/encode/httpx/issues/302#issuecomment-1368094190

**Reply by gbeane** [reply]
thanks.  this looks like it will work as long as we can drop our support for Python 3.9 (it seems like truststore is not available for anything older than 3.10)

I think 3.9 users can use SSL_CERT_FILE and point it at the ZScaler root certificate to work around the issue
